src/content/developers/tutorials/optimism-std-bridge-annotated-code/index.md

```diff
@@ -0,0 +1,1398 @@
+     * Variables *
+     *************/
+
+    // Messenger contract used to send and recieve messages from the other domain.
+    address public messenger;
+
+    /***************
+     * Constructor *
+     ***************/
+
+    /**
+     * @param _messenger Address of the CrossDomainMessenger on the current layer.
+     */
+    constructor(address _messenger) {
+        messenger = _messenger;
+    }
+```
+
+The one parameter that the contract needs to know, the address of the cross domain messenger on this layer.
+This parameter is set once, in the constructor, and never changes.
+
+```solidity
+
+    /**********************
+     * Function Modifiers *
+     **********************/
+
+    /**
+     * Enforces that the modified function is only callable by a specific cross-domain account.
+     * @param _sourceDomainAccount The only account on the originating domain which is
+     *  authenticated to call this function.
+     */
+    modifier onlyFromCrossDomainAccount(address _sourceDomainAccount) {
+```
+
+The cross domain messaging is accessible by any contract on the blockchain where it is running (either Ethereum mainnet or Optimism). 
+But we need the bridge on each side to *only* trust certain messages if they come from the bridge on the other side.
+
+
+```solidity
+        require(
+            msg.sender == address(getCrossDomainMessenger()),
+            "OVM_XCHAIN: messenger contract unauthenticated"
+        );
+```
+
+Only messages from the appropriate cross domain messenger (`messenger`, as you see below) can be trusted.
+
+```solidity
+
+        require(
+            getCrossDomainMessenger().xDomainMessageSender() == _sourceDomainAccount,
+            "OVM_XCHAIN: wrong sender of cross-domain message"
+        );
+```
+
+The way the cross domain messenger provides the address that sent a message with the other layer is [the `.xDomainMessageSender()` function](https://github.com/ethereum-optimism/optimism/blob/develop/packages/contracts/contracts/L1/messaging/L1CrossDomainMessenger.sol#L122-L128). 
+As long as it is called in the transaction that was initiated by the message it can provide this information.
+
+We need to make sure that the message we received came from the other bridge.
+
+```solidity
+
+        _;
+    }
+
+    /**********************
+     * Internal Functions *
+     **********************/
+
+    /**
+     * Gets the messenger, usually from storage. This function is exposed in case a child contract
+     * needs to override.
+     * @return The address of the cross-domain messenger contract which should be used.
+     */
+    function getCrossDomainMessenger() internal virtual returns (ICrossDomainMessenger) {
+        return ICrossDomainMessenger(messenger);
+    }
+```
+
+This function returns the cross domain messenger. 
+We use a function rather than the variable `messenger` to allow contracts than inherit from this one to use an algorithm to specify which cross domain messenger to use. 
```

Comment:
```suggestion
We use a function rather than the variable `messenger` to allow contracts that inherit from this one to use an algorithm to specify which cross domain messenger to use. 
```